### PR TITLE
Rework Cypress test "Register user, change password, login with new password".

### PR DESCRIPTION
**Pull request by dvkruchinin**
### Motivation and context
The test uses the credential generation function. It may match so that the generated password will not meet the CVAT requirements. In this case, the test will fail.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by bsekachev** [comment]
@dvkruchinin 


Hi, what do you think about my proposal on email? (removing created users after the test)
And can we do it everytime regardless on test passing status (I mean remove all created users anyway if test was failed or not, like ``finally`` works in JS)

**Comment by dvkruchinin** [comment]
@bsekachev 
Unfortunately, cypress currently does not have full functionality for running commands after a failed test. There is an experimental functionality. But I'm not sure we should use it right now. I added the ability to delete user accounts if the test uses user registration. Perhaps this option will add convenience to testing.
In any case, can use spec ```tests/cypress/integration/remove_users_tasks_projects.js``` to delete user accounts, created tasks, or projects.

File: tests/cypress/support/commands.js [hunk 1]
```diff
@@ -41,6 +41,40 @@ Cypress.Commands.add('userRegistration', (firstName, lastName, userName, emailAd
...
+                let userId = user['id'];
+                if (userId !== 1) {
+                    cy.request({
+                        method: 'DELETE',
+                        url: `/api/v1/users/${userId}`,
+                        headers: {
+                            Authorization: `Token ${authKey}`,
+                        },
+                    });
+                }
```
Comment (bsekachev) [inline tests/cypress/support/commands.js:72]: I am not sure this is the best idea.

Locally we usually have different users and these users created some tasks, projects and annotations.
Running this test once we will loose all the data in such approach. I believe created users should be a parameter of this command.

**Review by bsekachev — CHANGES_REQUESTED** [review]

File: tests/cypress/support/commands.js [hunk 2]
```diff
@@ -41,6 +41,40 @@ Cypress.Commands.add('userRegistration', (firstName, lastName, userName, emailAd
...
+    cy.request({
+        method: 'POST',
+        url: '/api/v1/auth/login',
+        body: {
+            username: Cypress.env('user'),
+            email: Cypress.env('email'),
+            password: Cypress.env('password'),
+        },
+    }).then(async (responce) => {
+        const authKey = await responce['body']['key'];
```
Comment (bsekachev) [inline tests/cypress/support/commands.js:54]: Is this operation really ``async``? Should we use ``await``?

File: tests/cypress/support/commands.js [hunk 3]
```diff
@@ -41,6 +41,40 @@ Cypress.Commands.add('userRegistration', (firstName, lastName, userName, emailAd
...
+        cy.request({
+            url: '/api/v1/users',
+            headers: {
+                Authorization: `Token ${authKey}`,
+            },
+        }).then(async (responce) => {
+            const responceResult = await responce['body']['results'];
```
Comment (bsekachev) [inline tests/cypress/support/commands.js:61]: The same question

Comment (bsekachev) [inline tests/cypress/support/commands.js:61]: Probably we should use linear promise chain here (now you use nested promise handlers)):

```js
cy.request({
       ....
    }).then(() => {
       .....
       return cy.request({ ... })  
    }).then(() => {
       .......
       delete users in circle
    })
```

File: tests/cypress/support/commands.js (hunk 1 above)
Comment (dvkruchinin) [reply tests/cypress/support/commands.js]: Oh, I understood. Modified the command to use the username to delete as a parameter.

File: tests/cypress/support/commands.js (hunk 2 above)
Comment (dvkruchinin) [reply tests/cypress/support/commands.js]: Yes, this is not necessary here.

File: tests/cypress/support/commands.js (hunk 3 above)
Comment (dvkruchinin) [reply tests/cypress/support/commands.js]: Answered above.

Comment (dvkruchinin) [reply tests/cypress/support/commands.js]: Rework in a slightly different form.

File: tests/cypress/support/commands.js [hunk 4]
```diff
@@ -41,6 +41,43 @@ Cypress.Commands.add('userRegistration', (firstName, lastName, userName, emailAd
...
+        url: '/api/v1/auth/login',
+        body: {
+            username: Cypress.env('user'),
+            email: Cypress.env('email'),
+            password: Cypress.env('password'),
+        },
+    }).then((responce) => {
+        const authKey = responce['body']['key'];
+        cy.request({
+            url: '/api/v1/users',
```
Comment (bsekachev) [inline tests/cypress/support/commands.js:56]: Only ten results will be returned. Use ``?page_size=all`` to get all the users.

Comment (dvkruchinin) [reply tests/cypress/support/commands.js]: Added.

**Review by bsekachev — APPROVED** [review]

**Pull request merged**